### ISSUE TITLE
allele frequency bars do not match data

### ISSUE THREAD

**Issue by bwlang** [igvteam/igv#1100]
<img width="1513" alt="image" src="https://user-images.githubusercontent.com/61636/152692648-d77a64d5-e6fa-47ff-8fee-f38cdb9aed62.png">

I've tried turning on and off downsampling - no effect.
Oddly - setting allelle frequency threshold from 0.1 to 0.5 also had no effect. Setting it to 0.9 does remove the colored bars.

I'll be uploading these some of these bams to SRA soon and can provide links if that's helpful.
maybe this is related to the mystery #967 

saw this in 2.11 and after upgrading to 2.12

**Comment by bwlang** [comment]
another surprising locus:
<img width="1016" alt="image" src="https://user-images.githubusercontent.com/61636/152693188-d9bf7bd1-7ef5-4e08-b241-b087313e1220.png">

I think the problem arises  because of the axis scale being limited to 1000x... I guess it should not sort by allele until after the depth threshold cuttoff application.

**Comment by jrobinso** [comment]
@bwlang Yes a link to a test data file and exactly locus to check would be helpful.

I don't understand the second comment.  "I guess it should not sort by allele until after the depth threshold cuttoff application"

**Comment by bwlang** [comment]
Hi Jim:

I think the problem is that the axis is being cut at 1000.
Let’s imagine that the full depth is 10000 and the al allele frequency is 10%.

That locus will appear to have a 100% alt-allele frequency if all the colored pixels are between 0 and 1000.

It would better for that to look like a 10% frequency with the depth threshold cut off or not I think.
That way the bar will always represent the data.

Another way to handle this might be to show the allele frequency as a horizontal percentage instead of a vertical one.

I don’t think people expect to “zoom in” on low frequency alleles by setting an data range - but I might be mistaken.


brad


On Feb 6, 2022, at 6:14 PM, Jim Robinson ***@***.******@***.***>> wrote:

EXTERNAL SENDER



@bwlang<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbwlang&data=04%7C01%7Clanghorst%40neb.com%7Cc29e9641dddb464e4d9708d9e9c674c9%7C77cefbc6b3d64d6a9f740664881c384b%7C0%7C0%7C637797860861792644%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=MGJw%2BVWevJEhCL7bPV2RnT8ZaeNeJcrEz4%2F%2BkGGKCvA%3D&reserved=0> Yes a link to a test data file and exactly locus to check would be helpful.

I don't understand the second comment. "I guess it should not sort by allele until after the depth threshold cuttoff application"

—
Reply to this email directly, view it on GitHub<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Figvteam%2Figv%2Fissues%2F1100%23issuecomment-1030936506&data=04%7C01%7Clanghorst%40neb.com%7Cc29e9641dddb464e4d9708d9e9c674c9%7C77cefbc6b3d64d6a9f740664881c384b%7C0%7C0%7C637797860861792644%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=rNf3agvs9ea4BmIPxvIsrAphwuZjh8MtxIDBh9GoOdk%3D&reserved=0>, or unsubscribe<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAAPBRDZVCV43W6XKFNXXJLUZ36GDANCNFSM5NVSSUCA&data=04%7C01%7Clanghorst%40neb.com%7Cc29e9641dddb464e4d9708d9e9c674c9%7C77cefbc6b3d64d6a9f740664881c384b%7C0%7C0%7C637797860861792644%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Ie1USfPT%2BX0A8futWugtbJapvvUPIkg0FXZ%2B3epx6%2Bs%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Clanghorst%40neb.com%7Cc29e9641dddb464e4d9708d9e9c674c9%7C77cefbc6b3d64d6a9f740664881c384b%7C0%7C0%7C637797860861792644%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=QDQgpL5EDOolESaYeywGjAcWw9qvPViH7mq1rDzjwRY%3D&reserved=0> or Android<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Clanghorst%40neb.com%7Cc29e9641dddb464e4d9708d9e9c674c9%7C77cefbc6b3d64d6a9f740664881c384b%7C0%7C0%7C637797860861792644%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=F6NhBuua0xLNocuJCKHpF2%2BKricRJArQ45U2zfjOUds%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Brad Langhorst, Ph.D.
Applications and Product Development Group Leader
New England Biolabs
***@***.******@***.***>

**Comment by jrobinso** [comment]
@bwlang OK I see what you are saying now.    I'll see what can be done.   BTW why is the scale hardcoded to 1,000? 

This part of IGV has been like it is since 2009.   Changing the vertical allele representation to account for thresholded values is doable, or I will investigate anyway,  however there would be a high bar to changing to something completely different, like horizontal slices.

**Comment by bwlang** [comment]
With multiplex amplicon sequencing the most critical factor is the depth of coverage of the lowest PCR product.
We often see amplicons that are at 20000x in the same run as ones at 20x (that’s not a goal - but is a reality of a changing virus).  A log scale flattens everything too much, making 20x and 200x look about the same. 
Fixing the axis at at a lower level allows one to see the relative values of the low coverage areas - but screws up perception of the allele frequency. 

Maybe this is a potential new mode: “show allele frequency as percentage truncated bars" (a bit wordy - hmm) . That would look the same if the whole bar is visible or log scale i guess (not sure if you log-transform the low variants and warp those more???).

Other ideas welcome! Or people can just remember to check if the huge swath of blue is 1% or 90% C.
You only get burned once or twice ;)

Some examples - in case they are helpful:

PS - it's so nice that one can multi-select all the coverage tracks and set them at once - I didn't figure that out for an embarrassingly long time ;)

log scale (ok - but coverage looks even - it's not):
<img width="1217" alt="image" src="https://user-images.githubusercontent.com/61636/152784938-a2636396-72fe-4f7a-b9d1-83129bad5d63.png">

linear autoscale group (can't see the difference between low regions at all):
<img width="1215" alt="image" src="https://user-images.githubusercontent.com/61636/152785177-448edafd-83ae-4bf6-aba1-50479b43e688.png">

linear 100x (best coverage picture for this marginal dataset? but allele frequencies are not very interpretable):
<img width="1200" alt="image" src="https://user-images.githubusercontent.com/61636/152785645-b8046c0d-e43c-472b-aed5-b6309e901701.png">

**Comment by bwlang** [comment]
wow - so quick - i look forward to the next release to try this out!

**Comment by jrobinso** [comment]
Should be out tomorrow, v2.12.1.  If it doesn't address your issue re-open
it.
